ui/components/ui/main-layout/main-layout.tsx

```diff
@@ -1,18 +1,101 @@
 "use client";
 
+import { usePathname } from "next/navigation";
+
 import { useSidebar } from "@/hooks/use-sidebar";
 import { useStore } from "@/hooks/use-store";
 import { cn } from "@/lib/utils";
 
+import { Navbar } from "../nav-bar/navbar";
 import { Sidebar } from "../sidebar/sidebar";
+
+// Page metadata mapping
+const PAGE_METADATA = [
```

Comment:
I might agree with extracting this as a shared config, but I’d prefer to handle it directly in the Next.js layout instead of introducing additional client components along the way. 

The function itself could live in a helper file (we’d need to decide which one), and then simply be called from the layout. 

Also, right now the page load doesn’t depend on the user, the user info is independent from the rendered components, is still that way?